### PR TITLE
Making telescope model names consistent all over the modules

### PR DESCRIPTION
**Pull request by RaulRPrado**
This PR does not add any new module. Many small changes were made in order to make the telescope model name consistent.

Before, the telescope model names included site (e.g. North-LST-1) in some places, and did not include it in some other places (e.g. in the array_model, where the site is given only once for the whole array).

Now, site is always given separate from the telescope model name. In the new convention, telescopeModelNames contain only telescope class (SST, MST or LST) and the specific identifier (e.g. MST-FlashCam-D).

There are few functions in db_handler that still uses the telescopeModelName with site, because they communicate directly with the DB. In these cases, the variable name was changed to telescopeModelNameDB.

The names of the variables were made consistent as well. Now telescopeModelName is used everywhere for the name of the model and telescopeName is used at the layout, to indicate the name of a certain telescope in the array (L-01, M-05 ...).

The variables for the model version were all replaced by modelVersion, instead of just version, for clarity. 

The attribute 'telescopeName' of TelescopeModel was changed by just 'name'. 

Some small refactoring were done in some places, e.g. db_handler.

### CODE REVIEW

File: simtools/db_handler.py [hunk 1]
```diff
@@ -492,10 +521,6 @@ def _getSiteParametersYaml(self, site, version, onlyApplicable=False):
         dict containing the parameters
         '''
 
-        if site.lower() not in ['north', 'south']:
-            raise ValueError('Site must be "North" or "South" (not case sensitive)')
-        site = 'lapalma' if 'north' in site.lower() else 'paranal'
```
Comment (orelgueta) [inline simtools/db_handler.py:497]: Why is this removed? I think it's needed to read the site variables (they are all in the same file and the only difference is the prefix lapalma/paranal).

**Review by orelgueta — APPROVED** [review]
All looks good. I left a few minor comments, only one discusses a potential break, but I approve anyway in case I am wrong.

File: simtools/db_handler.py (hunk 1 above)
Comment (RaulRPrado) [reply simtools/db_handler.py:497]: That was a mistake. I meant to remove the validation only and ended up removing the following line too. I fixed it now.

File: simtools/db_handler.py [hunk 2]
```diff
@@ -211,30 +219,45 @@ def getModelParameters(
         dict containing the parameters
         '''
 
-        _version = version
-        if version in ['Current', 'Latest']:
-            _version = self._getTaggedVersion(DatabaseHandler.DB_CTA_SIMULATION_MODEL, version)
+        _version = modelVersion
```
[the remark this answers is not in the document]
Comment (RaulRPrado) [reply simtools/db_handler.py:222]: Done!

File: simtools/db_handler.py [hunk 3]
```diff
@@ -211,30 +219,45 @@ def getModelParameters(
...
+                _telModelNameValidated,
+                _versionValidated,
                 runLocation,
                 onlyApplicable
             )
             return _pars
         else:
-            return self._getModelParametersYaml(telescopeName, version, onlyApplicable)
+            return self._getModelParametersYaml(
+                site,
```
[the remark this answers is not in the document]
Comment (RaulRPrado) [reply simtools/db_handler.py:244]: That was a mistake. Fixed now!

**Comment by RaulRPrado** [comment]
> All looks good. I left a few minor comments, only one discusses a potential break, but I approve anyway in case I am wrong.

Ok, thanks.

I fixed the mistakes you pointed out and will merge it now.
